client/extensions/woocommerce/components/query-locations/index.js

```diff
@@ -0,0 +1,65 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { fetchLocations } from 'woocommerce/state/sites/locations/actions';
+import { areLocationsLoaded } from 'woocommerce/state/sites/locations/selectors';
+
+class QueryLocations extends Component {
+	static propTypes = {
+		fetchLocations: PropTypes.func,
+		loaded: PropTypes.bool,
+		siteId: PropTypes.number.isRequired,
+	};
+
+	componentWillMount = () => {
+		const { siteId, loaded } = this.props;
+
+		if ( siteId && ! loaded ) {
```

Comment:
Is this checking for fetching state, or if something exists? If it's the latter you likely don't want it since we'll end up with stale data if a user revisits the same view without reloading the page.

Reply:
Its the former (was state fetched). That said, the data _can_ vary by site, but is highly unlikely to change on a given site (requires a plugin update to change.)

Reply:
So with a QueryComponent the general pattern is:
- Let the component render with current available data (which may be stale)
- But also fetch new data in 2 cases:
-- On Mounts
-- Or when Props change

The problem with not fetching when some state is loaded, is that we won't will fetch again until the user refreshes the page. (Or in the case of us persisting state, never again)

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/21297/commits/698f0228879b86ac921f12fd74df554610db7238